cashaddr.md

```diff
@@ -0,0 +1,117 @@
+# Address format for Bitcoin Cash
+
+Version 1.0, 2017-10-13
+
+## Abstract
+
+This document describes proposed address format to be used on Bitcoin Cash. It is a base32 encoded format using BCH codes as checksum and that can be used directly in links or QR codes.
+
+This format reuse the work done for Bech32[[1]](#bip173) and is similar is some aspects, but improves on others.
+
+## Specification
+
+The address is composed of
+1. A prefix indicating the network on which this address is valid.
+2. A separator, always `:`
+3. A base32 encoded payload indicating the destination of the address and containing a checksum.
+
+### Prefix
+
+The prefix indicates the network on which this addess is valid. It is set to `bitcoincash` for Bitcoin Cash main net, `xbctest` for bitcoin cash testnet and `xbcreg` for bitcoin cash regtest.
+
+The prefix is followed by the spearator `:`.
+
+When presented to users, the prefix may be omitted as it is part of the checksum computation. This ensure that addresses on different networks will remain incompatible, even in the absence of an explicit prefix.
+
+### Payload
+
+The payload is a base32 encoded stream of data.
+
+|     | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 |
+| --: | - | - | - | - | - | - | - | - |
+|  +0 | q | p | z | r | y | 9 | x | 8 |
+|  +8 | g | f | 2 | t | v | d | w | 0 |
+| +16 | s | 3 | j | n | 5 | 4 | k | h |
+| +24 | c | e | 6 | m | u | a | 7 | l |
+
+The payload is composed of 3 elements:
+1. A version byte indicating the type of address.
+2. A hash.
+3. A 40 bits checksum.
+
+#### Version byte
+
+The version byte's most signficant bit is reserved and must be 0. The 4 next bits indicate the type of address and the 3 least significant bits indicate the size of the hash.
+
+| Size bits | Hash size in bits |
```

Comment:
Since the paragraph above mentions that the type bits come first, it would be smoother to put the type bits explanation & table first instead of the hash bits.